src/rebase.js

```diff
@@ -116,13 +116,22 @@ module.exports = (function(){
   function _fetch(endpoint, options){
     _validateEndpoint(endpoint);
     optionValidators.context(options);
-    optionValidators.then(options);
     options.queries && optionValidators.query(options);
     var ref = firebase.database().ref(endpoint);
     ref = _addQueries(ref, options.queries);
-    ref.once('value', (snapshot) => {
+    return ref.once('value').then(snapshot => {
```

Comment:
Keep it. It's absolutely OK to return rejection here. It properly represents what happened — the fetch failed. And there's no such thing as _unhandled rejection_ — it's user's responsibility to `.catch` rejected Promise that you return.